storage/src/nodestore.rs

```diff
@@ -1031,6 +1051,7 @@ pub struct ImmutableProposal {
 impl ImmutableProposal {
     /// Returns true if the parent of this proposal is committed and has the given hash.
     #[must_use]
+    #[allow(clippy::needless_pass_by_value)]
```

Comment:
please use `expect`. I think this lint was disabled, so rebasing on main may mean you can delete this line.